Add login auth feature and db setup (progressing...)

nguyentntram opened the pull request:
- Connected MongoDB using mongooose
- Created authController.js for handing register/login/logout/verify/reset/...
- Setup Nodemailer
- Modify package.json in both client and server
- More features comming soon...

PhuocOng commented:
Sau em viết PR description chi tiết hơn nhé, ví dụ
**WHAT I DID:**
- Connected MongoDB using mongooose
- Created authController.js for handing register/login/logout/verify/reset/...
- Setup Nodemailer
- Modify package.json in both client and server
- More features comming soon...

**NEW API ENDPOINT**:
- /login
- /logout
- /otp..?

nguyentntram commented:
_(Hnhu push code lên branch cũ thì đâu đc tạo pr mới đúng k ạ, e tìm mãi mà k thấy chỗ để tạo pr mới trong branch của e :<< nên e viết description cho mấy cái e push lần 2 ở comment nhe, còn nếu tạo đc thì mn chỉ e nhé)_

**2nd PR: Update frontend and backend (Authentication)**
**1. BACKEND(server):** 
- Added reset password functionality: implemented reset feature in authController.js and created new route in authRoute.js
- Created userController.js and userRoute.js to handle fetching authenticated user data 

**2. FRONTEND (client):**
- Restructed folder layout: e có tách riêng các phần ra để dễ quản lí hơn í, kiểu ban đầu phần Greeting (chắc giống Homepage) nằm ở trong App.jsx, thì e nghĩ App nên để quản lý chủ yếu các route path thui. Còn phần Greeting kia thì e tạo folder riêng (có cả css và js)
- Created navbar: for logo, route link, login button,... (e thấy nhiều web đều có phần navbar trên đầu í nên e để thêm cho đẹp chứ nó cx k có ý nghĩa j lắm:>>)
- Created Login.jsx (for UI authentication) (in progress...)
- Added some img to assets folder (logo, icon,... này e lụm trên mạng với lấy đại ảnh react có sẵn thui)
